### ISSUE TITLE
How to run code on other dataset 

### ISSUE THREAD

**Issue by vedantpatel41** [ywyue/RoomFormer#27]
hello there,  i am interested in applying this model on my dataset. i have point cloud data of an building floor. how can i run this model on it? 
can you elaborate how can i do this.

**Comment by vedantpatel41** [comment]
well i have experimented  with my point cloud data, i have first created the density maps and ran the inference on them. here are the density images and the results. 
![03256](https://github.com/user-attachments/assets/965e299a-041c-450c-95d2-b409e26f56e1)

result: 
![image](https://github.com/user-attachments/assets/80e26644-5dcf-480a-b5a7-31d780092e42)

image: 

![03251](https://github.com/user-attachments/assets/fa0a1ba8-7b57-477e-ad4b-4d1897bd7d68)

results: 
![image](https://github.com/user-attachments/assets/d02ddb99-a564-41d3-906e-34829ad5847c)
 
 
i have tried to create the density image using the method in the code itself:  
![image](https://github.com/user-attachments/assets/6f44e31b-7e36-4a44-84e5-27d77eea0c46)
![image](https://github.com/user-attachments/assets/4a4f4db2-e24e-4acf-aac0-b562736da2f7)


the results i am getting are not good.  can you tell me what i can do to improve this results. also if i want to train this model on my dataset then how should i create the training data ? i mean how can i annotate it ?  it will be great help if you can answer this.

**Comment by ywyue** [comment]
Hi @vedantpatel41, 

RoomFormer was trained on Structured3D, a synthetic dataset of residential apartment scenes. We also had a variant of model trained on SceneCAD, a dataset of single-room scenes. Due to domain gap with your dataset, they may not work well, just as you show. Regarding the density map generation, did you follow the code for Structured3D or SceneCAD? Please observe which dataset is most similar to yours and use the corresponding model and density map preprocessing procedure.

To train the model on your dataset, you need to prepare the training pairs: density map + annotation. The density map can be obtained by projecting 3D point clouds to 2D using our code. The annotation is a list of polygons (which in turn are a list of ordered vertices), which needs to be collected by hand. I had no experience for annotating this. But one potential way to do that is to load point clouds to CouldCompare (or other PC viewer), check the coordinates of room corners and read them out sequentially.

**Comment by zxbzhineng** [comment]
@ywyue @vedantpatel41 Sorry to bother you, I want to make inferences on my own data, how should I proceed? Thank you

**Comment by zhangwanyao** [comment]
I see that Structured3D is the depth map and picture of indoor scene collected, how do you generate the point cloud data? I have seen the annotations of Structured3D, test, train, val are all annotated, corresponding to each room and type, how does this correspond? Looking forward to your reply. @ywyue

**Comment by zhangwanyao** [comment]
@vedantpatel41 Have you marked it? I am also working on an attempt to generate structured CAD directly from point cloud data.

**Comment by zhangwanyao** [comment]
![myplot](https://github.com/user-attachments/assets/0f099980-4e7b-4c46-af49-a441a56981d6)
@vedantpatel41 Have you marked it? I am also working on an attempt to generate structured CAD directly from point cloud data.I found that the labels below the test data set have been partitioned. After manual partitioning, the precise CAD drawings can be generated by reasoning. You can label according to the data format of the test data set.

**Comment by ywyue** [comment]
Hi @zxbzhineng, to make inferences on your own data using the pretrained network, you need to first convert your 3D scan to density map. Here is the code for preprocessing: https://github.com/ywyue/RoomFormer/tree/main/data_preprocess

If your scan looks more similar with Structured3D, please follow the preprocessing step of Structured3D, otherwise SceneCAD.

The inference code is here: https://github.com/ywyue/RoomFormer/blob/33ad087f50f15fce04719828db8d2a56fd4b630f/engine.py#L205 

You need to modify the code a bit to adapt it to your dataset.
